### PR TITLE
DOM test class changes

### PR DESCRIPTION
**Pull request by dphiffer**
Adds a "change classes" aspect to DOM tests: add a class or remove a class.

Example: "turn the nav donate button pink"
Screenshot 2024-11-22 at 12 01 43 PM

### CODE REVIEW

File: src/split-tests.js [hunk 1]
```diff
@@ -58,14 +58,27 @@ export default function split_tests_init() {
...
                     }
                 }
             }
 
+            if (variant.classes) {
+                for (let change of variant.classes) {
+                    let targets = document.querySelectorAll(change.selector);
+                    for (let target of targets) {
+                        if (change.change == 'add') {
+                            target.classList.add(change.class);
```
Comment (BatMiles) [inline src/split-tests.js:74]: I'm missing something - wouldn't this code mean that the added/removed classes were _always_ added/removed?

**Review by BatMiles — APPROVED** [review]
Tested locally, looks good! Left one code question

File: src/split-tests.js (hunk 1 above)
Comment (dphiffer) [reply src/split-tests.js:74]: Well it'll happen every time the variant that lands on the page with a class add/removal, but that will only happen some of the time. fwiw DomTests.php:76 is where that random selection happens.

Comment (BatMiles) [reply src/split-tests.js:74]: got it, thanks!

**Pull request merged**